src/GuzzleClient.php

```
@@ -0,0 +1,124 @@
+            throw new \Exception('Authentication failure');
+        }
+
+        // Handle internal server error
+        if ($response->getStatusCode() === 500) {
+            $body = (string) $response->getBody();
+            // Log or output error for debugging (as per reference implementation)
+            if ($this->logger !== null) {
+                $this->logger->error('Internal server error', ['body' => $body]);
+            } else {
+                error_log('Internal server error: ' . $body);
+            }
```

PHPMD (2.15.0)

The method request uses an else expression. Else clauses are basically not necessary and you can simplify the code by not using them. (undefined)

(ElseExpression)